src/GitVersion.Core.Tests/IntegrationTests/VersionInMergedBranchNameScenarios.cs

```diff
@@ -26,6 +26,21 @@ public void DoesNotTakeVersionFromNameOfNonReleaseBranch()
         fixture.AssertFullSemver("1.1.0-alpha.5");
     }
 
+    [TestCase("release")]
+    [TestCase("hotfix")]
+    public void DoesNotTakeVersionFromNameWhenItHasBeenAccidentalSpecifiedInBranch(string branch)
```

Comment:
This is a perfect test to showcase why this PR is needed. 👍🏼  I just think the method name can be simplified a bit.

```suggestion
    public void DoesNotTakeVersionFromBranchWithAccidentalVersion(string branch)
```